apps/ios/Shared/Views/Database/DatabaseErrorView.swift

```diff
@@ -59,29 +59,58 @@ struct DatabaseErrorView: View {
             }
         }
         .padding()
-        .frame(maxHeight: .infinity)    }
+        .frame(maxHeight: .infinity)
+    }
+
+    private func databaseKeyField() -> some View {
+        DatabaseKeyField(key: $dbKey, placeholder: "Enter passphrase…", valid: validKey(dbKey), onSubmit: runChat)
+    }
 
     private func saveAndOpenButton() -> some View {
         Button("Save passphrase and open chat") {
             if setDatabaseKey(dbKey) {
                 storeDBPassphraseGroupDefault.set(true)
                 initialRandomDBPassphraseGroupDefault.set(false)
             }
-            do {
-                try initializeChat(start: m.v3DBMigration.startChat, dbKey: dbKey)
-            } catch let error {
-                logger.error("initializeChat \(responseError(error))")
-            }
+            runChat()
```

Comment:
why this part is not in onSubmit?
```
if setDatabaseKey(dbKey) {
    storeDBPassphraseGroupDefault.set(true)
    initialRandomDBPassphraseGroupDefault.set(false)
}
```
I still get my password saved somehow even on submit, but not sure why